### ISSUE TITLE
[New build - DEFAULT] 24w29c: Hotfixes

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#5891]
Harvester now has a fule-type filter. Convenient to dump oxygen to not stall a H2 farm for example.

Hotfixes:
#5890 Warp crystal cant be mined
#5888 Placing a handheld on an item stand freezes game
#5886 Kiosk can freeze

**Comment by Crimson-Artist** [comment]
I have the fuel scoop set to expel water and the resulting cloud looks just like hydrogen gas. functionally it expelled water but is there any plans to add liquid physics so water pours out of the fuel scoop block or will everything that comes out will be a different color cloud?
![20240719021222_1](https://github.com/user-attachments/assets/01a50b48-cd1d-4f3a-ac5b-cdd508f835e3)

**Comment by asanagisae** [comment]
@tsunamayo, I don't believe O2 was being "stalled" on my ship. The O2 levels on the electrolyzer were constantly going up (being produced) and down (being moved to storage). I had an O2 tank that correctly received O2 while the electrolyzer was working.

If it were "stalled", then at some point the electrolyzer would hit maximum volume on O2 and then stop.

This did not happen. The electrolyzer never stopped producing H2 and O2. It never stopped consuming water. The O2 was directed to the O2 tank correctly.

The issue is that the H2 vanished near-instantly upon leaving the eletrolyzer.

It's possible that the LC engine is consuming the hydrogen at an unexpected rate. It's true that the H2 farming bases I built did not have LC engines connected in the fuel system. The ships _do_ have LC engines. That seems to be a major distinction.

I can do some testing but I may have lost my ship in a distant star system. My character died with no warning, possibly due to life support ending with no warning or indication of how much time I had left. So it may take a while to get things set up again.

**Comment by asanagisae** [comment]
I ran into an issue trying out the fuel scoop ejecting gas.

One, when set to eject, it will do so even if a logic block has turned it off.

Two, I set it to eject oxygen, but the type will display hydrogen when the amount of gas in it is zero.

https://github.com/user-attachments/assets/a267fb00-fb0e-47d7-aa00-d6d4a7377194
